docs/spec_draft.md

```diff
@@ -305,7 +309,32 @@ the IEEE-754 specification. For boolean element type, the behavior is same as
 // %result: [[6, 8], [10, 12]]
 ```
 
-&nbsp;[More Examples](../stablehlo/tests/interpret_add.mlir)
+# stablehlo.after_all
+
+### Semantics
+Ensures that the operations producing the `inputs` are ordered before any
+operation that depend on `result`.
```

Comment:
Proposed rewording:

```
Ensures that the operations producing the `inputs` are executed before any
operations that depend on `result`. Execution of this operation does nothing,
it only exists to establish data dependencies from `result` to `inputs`.
```

Feedback:
  1) "are ordered" => "are executed" given the discussion above on the definition of the token type.
  2) "any operation that depend" doesn't compute grammatically, so I proposed one possible fix.
  3) Explicitly calling out that execution of this operation does nothing.